types/studio.ts

```diff
@@ -30,4 +31,28 @@ export type StudioOutput = Partial<FullStudioOutput>;
 export interface StudioContext extends Context<StudioOutput> {
   studio: Studio;
   studioName: string;
+  // Server functions to lazy load extra studio data
+  $getLabels: () => Promise<Label[]>;
+  $getAverageRating: () => Promise<number>;
+  $getParents: () => Promise<Studio[]>;
+  $getSubStudios: () => Promise<Studio[]>;
+}
+
+// Merge the studio's initial data and previous plugins piped data
+export async function getMergedData(ctx: StudioContext): Promise<StudioOutput> {
+  const { data, studio } = ctx;
+
+  const initialData: StudioOutput = {
+    name: studio.name,
+    description: studio.description || undefined,
+    addedOn: studio.addedOn.valueOf(),
+    favorite: studio.favorite,
+    bookmark: studio.bookmark || undefined,
+    aliases: studio.aliases,
+    labels: (await ctx.$getLabels()).map((l) => l.name),
+    parent: (await ctx.$getParents()).map((p) => p.name)?.[0],
```

Comment:
real nitpicking, but you don't need to map if you're only going to use the first element
```suggestion
    parent: (await ctx.$getParents())?.[0]?.name,
```

Reply:
Removed those ;-)